plugins/inputs/docker/docker.go

```diff
@@ -82,7 +84,12 @@ var sampleConfig = `
   endpoint = "unix:///var/run/docker.sock"
 
   ## Set to true to collect Swarm metrics(desired_replicas, running_replicas)
+  ## Note: configure this in one of the manager nodes in a Swarm cluster.
+  ## configuring in multiple Swarm managers results in duplication of metrics.
   gather_services = false
+  ## If gather_services is true, only collect if this node is the leader.
+  ## This allows the same configuration to be used on all nodes in a Swarm.
+  detect_leader = false
```

Comment:
I think we should turn these bool pairs into a tri-bool type setting:
```
  ## Service mode sets the condition for collecting Swarm data.  Can be one of
  ## "leader", "always", or "never".
  swarm_reporting = "leader"
```

Leave the old option as is, but if the new option is not set translate it to "always" or "never".  When the new option is set ignore the old value.